src/meadowrun/local_code.py

```diff
@@ -11,47 +11,67 @@ def zip_local_code(
     result_zip_dir: str,
     include_sys_path: bool = True,
     additional_python_paths: Iterable[str] = tuple(),
-    additional_paths: Iterable[str] = tuple(),
+    additional_non_python_paths: Iterable[str] = tuple(),
     include_extensions: Iterable[str] = (".py",),
 ) -> Tuple[str, List[str], List[str]]:
-    """Zips ups files from various sources, consolidating paths.
-
-    Returns the path to the resulting zip file, the list of paths into the zip file that
-    correspond to given additional_python_paths and gathered from sys.path if
-    include_sys_path is enabled, and the corresponding list of paths into the zip file
-    that were given in additional_paths.
+    """
+    The goal of this function is to zip up "the current local python code excluding the
+    interpreter code" as well as any additional non-python files that might be needed to
+    run code remotely.
+
+    There are two types of folders/files that make it into the zip file. The first are
+    "python paths". Assuming include_sys_path is True, "python paths" will be
+    additional_python_paths + paths in sys.path that are NOT part of the interpreter,
+    i.e. paths that start with sys.prefix, etc. The second are "non-python paths". These
+    two types of paths are treated exactly the same in the zip file. The only difference
+    between these two types of paths is that we keep track of them separately in the
+    return value--the returned tuple will include a list of "python paths" as they
+    appear in the zip file and "non-python paths" as they appear in the zip file.
+
+    Non-python paths are useful for two reasons:
+    - One is to add structure to the zip file. This function will consolidate paths that
+      share a prefix. So for example, if `src/foo` and `src/bar/baz` are python paths,
+      but `src` itself is not a python path, you can set `src` as a non-python path and
+      your zip file will have the right structure. If you don't do this, `foo` and `baz`
+      will just be top-level directories in the zip file.
+    - You'll usually want to include the current working directory as a "non-python
+      path" if it's not already a python path. Then when you unzip, you can set the
+      working directory to the corresponding unzipped directory and relative paths will
+      work as expected.
+
+    Both python paths and non-python paths will only include *.py files (or whatever is
```

Comment:
This is just describing the current behavior, but it seems a bit odd--should we even be including the *.py files at all for non-python paths by default?

Reply:
most recent take:
- python paths (additional or sys.path) only zip .py files (and this is not changeable  in the run API until we see a real use case for it, might as well remove it here actually)
- non-python _files_ are defined by full wildcard globs, i.e. including any extensions. Equivalent of getcwd is `"./**/*.py"`
- btw, "non-python files" ugh, how about `additional_files`?

So we'd have: 
- `additional_python_paths`  which takes a list of paths and are put on PYTHONPATH remotely
- `additional_files` which takes a list of glob patterns